pkg/discovery/dtofactory/namespace_entity_dto_builder.go

```diff
@@ -0,0 +1,117 @@
+package dtofactory
+
+import (
+	"github.com/golang/glog"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/metrics"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/repository"
+	sdkbuilder "github.com/turbonomic/turbo-go-sdk/pkg/builder"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+)
+
+type namespaceEntityDTOBuilder struct {
+	NamespaceMap map[string]*repository.KubeNamespace
+}
+
+func NewNamespaceEntityDTOBuilder(namespaceMap map[string]*repository.KubeNamespace) *namespaceEntityDTOBuilder {
+	builder := &namespaceEntityDTOBuilder{
+		NamespaceMap: namespaceMap,
+	}
+	return builder
+}
+
+// Build entityDTOs based on the given node list.
+func (builder *namespaceEntityDTOBuilder) BuildEntityDTOs() ([]*proto.EntityDTO, error) {
+	var result []*proto.EntityDTO
+
+	for _, namespace := range builder.NamespaceMap {
+		// id.
+		namespaceID := namespace.UID
+		entityDTOBuilder := sdkbuilder.NewEntityDTOBuilder(proto.EntityDTO_NAMESPACE, namespaceID)
+
+		// display name.
+		displayName := namespace.Name
+		entityDTOBuilder.DisplayName(displayName)
+
+		// Resource commodities sold.
+		commoditiesSold, err := builder.getQuotaCommoditiesSold(namespace)
+		if err != nil {
+			glog.Errorf("Error creating commoditiesSold for %s: %s", namespace.Name, err)
+			continue
+		}
+
+		// VMPMAccess commodity sold.
+		namespaceAccessComm, err := sdkbuilder.NewCommodityDTOBuilder(proto.CommodityDTO_VMPM_ACCESS).
```

Comment:
can we make the WorkloadController Movable false instead. Or, we can disable Move during the probe registration

Reply:
Yes, this makes sense. We can make WorkloadController and Pod Movable false instead of constrained by the VMPM access commodity. 
And I also noticed in the existing logic in pod_entity_dto_builder, pod buys from VDC provider with movable as false to disable move across Namespaces.
I'll remove the VMPM access commodity here.